app/forms/csv_upload_form.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CsvUploadForm
+  include ActiveModel::Model
+
+  attr_reader :keyword_ids
+
+  validates :file, presence: true, csv_keyword_file: true
+
+  def save(file)
+    @file = file
+    return false unless valid?
+
+    begin
+      @keyword_ids = Keyword.create(parsed_keywords).pluck('id')
+      true
+    rescue ActiveRecord::ValueTooLong
+      errors.add(:file, I18n.t('activemodel.csv.errors.invalid_keyword_length'))
```

Comment:
I'm curious, let's say this error is happening on the middle of the process, will all created keywords in this process get rollback? Should we add a test for that @Goose97 

```
# in the same file
1. valid_keyword_1 -> created
2. valid_keyword_2 -> created
3. exceed_255_keyword -> error 💥 
4. ...
```

will the **valid_keyword_1 and valid_keyword_2** got deleted?

Reply:
With the validation logic now all moved into the Validator class, we can omit the rescue block. In this case, I think we can safely ignore this scenario.

What do you think?

Reply:
We still need to test that keywords file doesn't get processed partially. Here's the test case for that 22fb4488dab5685eb7201d9fac14743512bc57c3